src/lib/logical_query_plan/join_node.cpp

```diff
@@ -106,89 +113,92 @@ std::shared_ptr<LQPUniqueConstraints> JoinNode::_output_unique_constraints(
 
   const auto join_predicate = std::dynamic_pointer_cast<BinaryPredicateExpression>(join_predicates().front());
   if (!join_predicate || join_predicate->predicate_condition != PredicateCondition::Equals) {
-    // Also, no guarantees implemented yet for other join predicates than _equals() (Equi Join)
-    return std::make_shared<LQPUniqueConstraints>();
+    // Also, no guarantees implemented yet for other join predicates than _equals() (Equi Join).
+    return std::make_shared<UniqueColumnCombinations>();
   }
 
-  // Check uniqueness of join columns
+  // Check uniqueness of join columns.
   bool left_operand_is_unique =
-      !left_unique_constraints->empty() &&
-      contains_matching_unique_constraint(left_unique_constraints, {join_predicate->left_operand()});
+      !left_unique_column_combinations->empty() &&
+      contains_matching_unique_column_combination(left_unique_column_combinations, {join_predicate->left_operand()});
   bool right_operand_is_unique =
-      !right_unique_constraints->empty() &&
-      contains_matching_unique_constraint(right_unique_constraints, {join_predicate->right_operand()});
+      !right_unique_column_combinations->empty() &&
+      contains_matching_unique_column_combination(right_unique_column_combinations, {join_predicate->right_operand()});
 
   if (left_operand_is_unique && right_operand_is_unique) {
-    // Due to the one-to-one relationship, the constraints of both sides remain valid.
-    auto unique_constraints =
-        std::make_shared<LQPUniqueConstraints>(left_unique_constraints->begin(), left_unique_constraints->end());
-    std::copy(right_unique_constraints->begin(), right_unique_constraints->end(),
-              std::back_inserter(*unique_constraints));
-    return unique_constraints;
+    // Due to the one-to-one relationship, the UCCs of both sides remain valid.
+    auto unique_column_combinations = std::make_shared<UniqueColumnCombinations>(
+        left_unique_column_combinations->begin(), left_unique_column_combinations->end());
+    std::copy(right_unique_column_combinations->begin(), right_unique_column_combinations->end(),
+              std::inserter(*unique_column_combinations, std::next(unique_column_combinations->begin())));
```

Comment:
UCCs are stored in sets now. Thus, `inserter` rather than `back_inserter`. The iterator position was indeed a copy mistake, changed to `unique_column_combinations->end()`. However, this position is irrelevant for sets.